### DISCUSSION TITLE
First Key on Keyboard reloads Organ

### DISCUSSION THREAD

**Discussion by JKBert** [GrandOrgue/grandorgue#1668] [Help]
### Discussed in https://github.com/GrandOrgue/grandorgue/discussions/527

<div type='discussions-op-text'>

<sup>Originally posted by **larspalo** September 16, 2021</sup>
Originally posted 2018-08-30 15:39:12.480000 by jreichels in GrandOrgue discussion forum on SourceForge<br><br>Dear Gents,

When I connect my Viscount organ keyboard via Midi to GO (Win 10 Pro), then the first note on Key 36 will reload the Organ-File. The other keys (37 to 96) are shifted one note. The Midi-Log said, Key Note 36 was pressed.

Have somebody an idea, what the problem is? Thanks for helping.

Best regards

Jreichels</div>

I have exactly the same problem with GO 3.13.0 (built from sources on a Linux Devuan box). I have tried to reinstall organ (default and Kassel), to delete cache and configuration files without success. Only one MIDI input is configured (keyboard).

Help will be welcome,

JB

**Comment by JKBert** [comment]
I have found there is a manual4 keyboard with this kind of events. Question is : how can I change parameters for this keyboard ? It doesn't appear in MIDI configuration...

**Comment by oleg68** [comment]
@JKBert Could you answer some questions?

>  then the first note on Key 36 will reload the Organ-File

Do you have anything in the MIDI column at File->Settings->Organs ?

> The other keys (37 to 96) are shifted one note. The Midi-Log said, Key Note 36 was pressed.

Does pressing the Key 37 causes logging that Key Note 36 is pressed?

> I have tried to reinstall organ (default and Kassel), to delete cache and configuration files without success. 

Reinstalling GO does not delete configuration files. You have to delete the file `User home directory`\AppData\Roaming\GrandOrgueConfig manually and you may do it without reinstalling GrandOrgue.

**Comment by oleg68** [comment]
@JKBert Sorry, I can't troubleshoot the problem without additional information, so I transfer this issue to discussion.

**Comment by Piedone** [comment]
Sorry to wake up an ancient discussion, but I also have the same issue with the latest v3.16.2-1. Walking down the keyboard works up until the lowests key, which reloads the default organ. Here's a video of what happens:

https://github.com/user-attachments/assets/c0fdef04-6fb5-44c7-b018-c0356ef451b7

The keys appear to be logged correctly.

I have this in the Organs menu:

<img width="933" height="753" alt="image" src="https://github.com/user-attachments/assets/8cae14ae-9cbd-43a9-9d9a-b6295f6f1651" />

I'm not sure what the MIDI column means here, because I'm using my keyboard with the other two samples too and it works fine (except for this issue).

**Reply by davidgritter** [reply]
the midi column says 'yes' when the organ is programmed to load on a specific midi message clicking on the MIDI button will bring up the normal window for programming midi events, where you can change or delete this event to prevent the organ from loading from midi .  Most likely this has been programmed to respond to a NOTE ON  message from your keyboard.

**Reply by oleg68** [reply]
> Sorry to wake up an ancient discussion, but I also have the same issue with the latest v3.16.2-1. Walking down the keyboard works up until the lowests key, which reloads the default organ. Here's a video of what happens:

> I'm not sure what the MIDI column means here, because I'm using my keyboard with the other two samples too and it works fine (except for this issue).

Thank you. The screenshot is right. `Yes` in the `MIDI` column shows that some MIDI input events are configured for loading this organ.

Having positioned on the Demo organ in the tab, could you click on the `MIDI...` button and post the screenshot of the next dialog here?

**Reply by Piedone** [reply]
Thank you @davidgritter and @oleg68 for helping me.

Checking out the MIDI settings for the demo organ, I see this:

<img width="491" height="377" alt="image" src="https://github.com/user-attachments/assets/2da596c8-733f-48ac-8ac9-e05988c303c9" />

I remember that I needed to run through "Detect complex MIDI setup" here to get the keyboard working with GO.

Contrast it with the settings of any other (also working) organ: 
<img width="491" height="377" alt="image" src="https://github.com/user-attachments/assets/093d7b6c-d8a2-4bca-b537-5f1459312746" />

Should I change something here?

BTW an interesting thing: while I keep the Program Settings Menu open (not necessarily the Organs tab) I can play the lowest key just fine.

**Reply by oleg68** [reply]
You can see some MIDI key event assigned to loading the demo organ. Just delete replace the "Event" with "(none)" and click to "OK".

**Reply by Piedone** [reply]
Ah, that's it. Changing that fixed the issue. Thank you! I don't know why that was set, but I think when I initially set up GO I configured that mistakenly while trying to get the keyboard working.
